### PR TITLE
Edit project API endpoint

### PR DESCRIPTION
**Pull request by vhf**
#249

### CODE REVIEW

File: apis/core/bootstrap/shapes/cube-project.ts [hunk 1]
```diff
@@ -49,4 +49,33 @@ ${shape('cube-project/create')} {
...
+    ${sh.property} [
+      ${sh.name} "Project name" ;
+      ${sh.path} ${rdfs.label} ;
+      ${sh.minCount} 1 ;
+      ${sh.maxCount} 1 ;
+      ${sh.minLength} 1 ;
+      ${sh.order} 10 ;
+    ] ;
+    ${sh.property} [
+      ${sh.path} ${cc.csvMapping} ;
```
Comment (martinmaillard) [inline apis/core/bootstrap/shapes/cube-project.ts:66]: I would add `sh:minCount 1`, `sh:maxCount 1` and `sh:order 20` here

File: apis/core/lib/domain/cube-projects/update.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    dataset.delete($rdf.quad(project.term, rdfs.label, oldLabel))
+  }
+  if (newLabel) {
+    storedProject.pointer.addOut(rdfs.label, newLabel)
+  }
+
+  const newNamespace = resource.out(cc.csvMapping).out(cc.namespace).term
+  if (newNamespace) {
+    const csvMapping = await store.getResource<CsvMapping>(project.out(cc.csvMapping).term)
+    const csvMappingPointer = csvMapping?.pointer
```
Comment (martinmaillard) [inline apis/core/lib/domain/cube-projects/update.ts:42]: Isn't it enough to do `csvMapping.namespace = newNamespace`?

File: apis/core/lib/domain/cube-projects/update.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      for (const q of namespacesDataset) {
+        namespaces.push(q)
+      }
+
+      const namespace = namespaces[0]
+      if (namespace) {
+        csvMappingPointer.dataset.delete($rdf.quad(csvMappingPointer.term, cc.namespace, namespace.object))
+      }
+      csvMappingPointer.addOut(cc.namespace, newNamespace)
+    }
```
Comment (martinmaillard) [inline apis/core/lib/domain/cube-projects/update.ts:56]: Isn't it enough to do csvMapping.namespace = newNamespace?

Comment (vhf) [reply apis/core/lib/domain/cube-projects/update.ts:56]: Apparently so…

File: apis/core/lib/domain/cube-projects/update.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  const newNamespace = resource.out(cc.csvMapping).out(cc.namespace).term as NamedNode
+  if (newNamespace) {
+    const csvMapping = await store.getResource<CsvMapping>(project.out(cc.csvMapping).term)
+
+    if (csvMapping) {
+      csvMapping.namespace = newNamespace
+    }
+  }
```
Comment (tpluscode) [inline apis/core/lib/domain/cube-projects/update.ts:46]: I'd prefer the mutation be done by functions on the appropriate mixin classes from `apis/core/lib/domain/cube-projects/Project.ts` and `apis/core/lib/domain/csv-mapping/CsvMapping.ts`

**Review by tpluscode — CHANGES_REQUESTED** [review]
With #298 merged, will need to also add `cc:publishGraph` to the project's update shape

**Review by tpluscode — APPROVED** [review]

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**